es6-promise/es6-promise.d.ts

```diff
@@ -51,6 +51,7 @@ declare module Promise {
 	 * Make a promise that rejects to obj. For consistency and debugging (eg stack traces), obj should be an instanceof Error
 	 */
 	function reject(error: any): Promise<any>;
+	function reject<T>(error: T): Promise<T>;
```

Comment:
This is to be consistent with 
https://github.com/Microsoft/TypeScript/blob/master/lib/lib.core.es6.d.ts#L5237

I recently try to migrate code to ES6 target and without that it fail to compile in some cases. Have to manually specify type for reject anyway.